src/transformers/generation/utils.py

```diff
@@ -1332,18 +1332,15 @@ def _get_cache(self, cache_implementation: str, max_batch_size: int, max_cache_l
         Returns the resulting cache object.
         """
         cache_cls: Cache = NEED_SETUP_CACHE_CLASSES_MAPPING[cache_implementation]
+        if cache_implementation == "sliding_window":
+            max_cache_len = min(self.config.sliding_window, max_cache_len)
+
         need_new_cache = (
             not hasattr(self, "_cache")
             or (not isinstance(self._cache, cache_cls))
-            or self._cache.max_batch_size < max_batch_size
+            or self._cache.max_batch_size != max_batch_size
```

Comment:
(unrelated to the sliding window cache) this was incorrect, we need a new cache object with a different batch size

Reply:
that's a nice catch!